web/html/src/manager/virtualization/guests/console/guests-console.js

```diff
@@ -173,6 +189,11 @@ class GuestsConsole extends React.Component<Props, State> {
                  this.state.currentToken, "application/json").promise.then(
       response => {
         this.popupSubmit = undefined;
+
+        // Schedule token refresh a minute before its expiration
+        const tokenExp = getTokenExpiration(response);
+        setTimeout(this.refreshToken, tokenExp - Date.now() - 60000);
```

Comment:
What about cleaning this timeout (like in the interval case)? Wouldn't not cleaning lead to a infinite refresh loop? Like in this hypothetical scenario:
- I use the console for more than an hour, `refreshToken` here is called, resulting in setting the timeout
- I navigate away, the timeout still exists (I'm probably wrong here)
- After some time `refreshToken` is called. What happens?

I hope this is not the case, but still I think, it would be fair to clean the timeout too.

Reply:
right, that one needs to be cleaned up